streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -307,12 +355,16 @@ public synchronized void start() {
         log.debug("Starting Kafka Stream process");
 
         if (state == KafkaStreams.State.CREATED) {
-            setState(KafkaStreams.State.RUNNING);
```

Comment:
sure - i'll put it back. Didn't really make sense to me to set the state to running before the threads have started. but whatever